### PR TITLE
MAILET-123 Mailet documentation generation fails

### PR DESCRIPTION
**Pull request by aduprat**
I will add a commit to update the plugin page:
http://james.apache.org/mailet/maven-mailetdocs-plugin/index.html

### CODE REVIEW

File: dockerfiles/compilation/java-6/compile.sh [hunk 1]
```diff
@@ -43,14 +43,26 @@ git checkout $SHA1
...
 fi
 
-# Retrieve result
-
 if [ $? -eq 0 ]; then
    cp server/app/target/james-server-app-*-app.zip $DESTINATION
 fi
+
+## Generate the mailetdocs
+mvn org.apache.james:mailetdocs-maven-plugin:2.5.1-SNAPSHOT:aggregate
```
Comment (mbaechler) [inline dockerfiles/compilation/java-6/compile.sh]: what about deploying the plugin in apache nexus instead ?

Comment (aduprat) [inline dockerfiles/compilation/java-6/compile.sh]: As it is embedded in the project, we will never be able to compile/test it thru Jenkins:
we should deploy each release before.

File: mailet/src/site/xdoc/mailetdocs-maven-plugin/index.xml [hunk 2]
```diff
@@ -31,12 +31,16 @@
...
-                <p>MailetDocs plugin has one goal:</p>
+                <p>MailetDocs plugin has two goals:</p>
                 <ul>
                     <li>
                         <a href='mailetdocs-mojo.html'>mailetdocs:mailetdocs</a>
                         Generates catalog and reports on mailets and matchers
                     </li>
+                    <li>
+                        <a href='mailetdocs-mojo.html'>mailetdocs:aggregate</a>
+                        Generates catalog and reports on mailets and matchers on a project and all its subprojects
```
Comment (rouazana) [inline mailet/src/site/xdoc/mailetdocs-maven-plugin/index.xml]: That's hard to understand the difference with the previous goal. Can you modifiy the previous to tell its limits?

**Comment by aduprat** [comment]
test this please

File: mailet/src/site/xdoc/mailetdocs-maven-plugin/index.xml (hunk 2 above)
Comment (aduprat) [inline mailet/src/site/xdoc/mailetdocs-maven-plugin/index.xml]: Right

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

File: dockerfiles/compilation/java-6/compile.sh (hunk 1 above)
Comment (mbaechler) [inline dockerfiles/compilation/java-6/compile.sh]: I mean, we can deploy it by hand, it's not that critical right now.

Comment (aduprat) [inline dockerfiles/compilation/java-6/compile.sh]: If we are not giving the fully-qualified goal, maven will try to execute the last released version of the plugin (2.5.0).

Comment (mbaechler) [inline dockerfiles/compilation/java-6/compile.sh]: It's my point : you should release a version, upload it and get rid of the version here.

Comment (aduprat) [inline dockerfiles/compilation/java-6/compile.sh]: The point is when we will modify this mojo, we will not be able to test that it's working on Jenkins.
We should release, upload; if Jenkins fails due to error, then we should re-release; re-upload.

Comment (aduprat) [inline dockerfiles/compilation/java-6/compile.sh]: Wel, that's not so simple; as this project is part of the mailet sub-project and it depends on classes of mailet-api; i need to publish all mailets jars...
I don't know if we expect this yet.

**Comment by aduprat** [comment]
merged

**Pull request closed without merging**